From d0e432329d71eac3c95c3a8004932930339947b8 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Tue, 4 Jun 2019 15:32:36 +0800
Subject: [PATCH 3/3] MGS-4913: Fix eglBindAPI issue in EGL no_config test case

In EGL no_config test case, OpenVG is the last item, will bind current thread to EGL_OPENVG_API,
but no_config test does not recover current thread with the initial value, refer to eglBindAP spec,
https://www.khronos.org/registry/EGL/sdk/docs/man/html/eglBindAPI.xhtml
(The initial value of the current rendering API is EGL_OPENGL_ES_API)

dEQP-EGL.functional.query_context.simple.query_api is next in EGL test group,
which expect EGL_OPENGL_ES_AP but it is EGL_OPENVG_API, so causing problem.

this patch save old API before test EGL no_config, restore it after finish.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
(cherry picked from commit 745a592e864603459cefca9c74dd2b74a8dcbed5)
---
 modules/egl/teglCreateContextTests.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/egl/teglCreateContextTests.cpp b/modules/egl/teglCreateContextTests.cpp
index dbacd741f..493917214 100644
--- a/modules/egl/teglCreateContextTests.cpp
+++ b/modules/egl/teglCreateContextTests.cpp
@@ -125,6 +125,7 @@ public:
 		const eglw::Library&		egl		= m_eglTestCtx.getLibrary();
 		const eglu::UniqueDisplay	display	(egl, eglu::getAndInitDisplay(m_eglTestCtx.getNativeDisplay(), DE_NULL));
 		tcu::TestLog&				log		= m_testCtx.getLog();
+		const EGLenum	oldApi	= egl.queryAPI();
 
 		if (!eglu::hasExtension(egl, *display, "EGL_KHR_no_config_context"))
 			TCU_THROW(NotSupportedError, "EGL_KHR_no_config_context is not supported");
@@ -164,6 +165,11 @@ public:
 			}
 		}
 
+		if (oldApi != egl.queryAPI())
+		{
+			egl.bindAPI(oldApi);
+		}
+
 		return STOP;
 	}
 };
-- 
2.17.1

